src/quote/provider/icusd.ts

```diff
@@ -0,0 +1,151 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import { Address } from 'viem'
+
+import { TheUSDCYieldIndex } from 'constants/tokens'
+import {
+  FlashMintNavTransactionBuilder,
+  FlashMintWrappedBuildRequest,
+  WrappedTransactionBuilder,
+} from 'flashmint'
+import { getExpectedReserveRedeemQuantity } from 'utils/custom-oracle-nav-issuance-module'
+import { getUsdcBalance } from 'utils/erc20'
+
+import { FlashMintNavQuoteProvider } from '../flashmint/nav'
+import { WrappedQuoteProvider } from '../flashmint/wrapped'
+import { QuoteProvider, QuoteToken } from '../interfaces'
+import {
+  FlashMintContractType,
+  FlashMintQuote,
+  FlashMintQuoteRequest,
+} from '../provider'
+import { SwapQuoteProvider } from '../swap'
+import { buildQuoteResponse } from './utils'
+
+export interface IcUsdQuoteRequest extends FlashMintQuoteRequest {
+  chainId: number
+  isMinting: boolean
+  inputToken: QuoteToken
+  outputToken: QuoteToken
+  indexTokenAmount: BigNumber
+  // FlashMintNav always needs the input amount
+  inputTokenAmount: BigNumber
+  slippage: number
+}
+
+export class IcUsdQuoteRouter
+  implements QuoteProvider<IcUsdQuoteRequest, FlashMintQuote>
+{
+  icUsd = TheUSDCYieldIndex
+  constructor(
+    private readonly rpcUrl: string,
+    private readonly swapQuoteProvider: SwapQuoteProvider
+  ) {}
+
+  async getQuote(request: IcUsdQuoteRequest): Promise<FlashMintQuote | null> {
+    // TODO: remove in production; for testing only
+    if (request.isMinting) {
+      return await this.getFlashMintWrappedQuote(request)
+    }
```

Comment:
@edkim @pblivin0x This currently makes the tests work. So only minting with FMWrapped works. If you comment the lines above, you'll get the routing how it's supposed to be according to the doc (using FMNav for minting).

When I run this tho, I get this error:
<img width="792" alt="Bildschirmfoto 2024-10-26 um 18 25 32" src="https://github.com/user-attachments/assets/18d70344-0050-4af3-b89d-0dd9951925d0">
<img width="1433" alt="Bildschirmfoto 2024-10-26 um 18 25 23" src="https://github.com/user-attachments/assets/ae0dda35-7f2a-4eb7-9e12-5dc1b4032504">